cmds/exp/tcpdump/tcp.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/binary"
+	"fmt"
+	"strings"
+
+	"github.com/gopacket/gopacket/layers"
+)
+
+// tcpData returns a string representation of the TCP layer.
+func tcpData(layer *layers.TCP, length int) string {
+	var data string
+
+	flags := tcpFlags(*layer)
+	opts := tcpOptions(layer.Options)
+
+	data = fmt.Sprintf("Flags [%s], seq %d, ack %d, win %d, options [%s], length %d", flags, layer.Seq, layer.Ack, layer.Window, opts, length)
+
+	return data
+}
+
+// tcpFlags returns a string representation of the TCP flags.
+func tcpFlags(layer layers.TCP) string {
+	var flags string
+	if layer.PSH {
+		flags += "P"
+	}
+	if layer.FIN {
+		flags += "F"
+	}
+	if layer.SYN {
+		flags += "S"
+	}
+	if layer.RST {
+		flags += "R"
+	}
+	if layer.URG {
+		flags += "U"
+	}
+	if layer.ECE {
+		flags += "E"
+	}
+	if layer.CWR {
+		flags += "C"
+	}
+	if layer.NS {
+		flags += "N"
+	}
+	if layer.ACK {
+		flags += "."
+	}
+
+	return flags
+}
+
+// tcpOptions returns a string representation of the TCP options.
+func tcpOptions(options []layers.TCPOption) string {
+	var opts string
+
+	for _, opt := range options {
+		opts += tcpOptionToString(opt) + ","
+	}
+
+	return strings.TrimRight(opts, ",")
+}
+
+// tcpOptionToString returns a string representation of the TCP option.
+func tcpOptionToString(opt layers.TCPOption) string {
+	switch opt.OptionType {
+	case layers.TCPOptionKindMSS:
+		if len(opt.OptionData) >= 2 {
+			return fmt.Sprintf("%s val %v",
```

Comment:
I would prefer that we minimize this multi-line parameter usage. It's not really idiomatic and it makes using tools like grep really hard. Might you consider making code like this one line? We don't have a 80- or 96-char column limit in u-root.

Reply:
Yeah, good catch.
 I condensed it into an oneliner for both option types.